core/src/impl/Kokkos_Core.cpp

```diff
@@ -282,11 +295,206 @@ void initialize_profiling(const InitArguments& args) {
     std::exit(EXIT_SUCCESS);
   }
   Kokkos::Tools::parseArgs(args.tool_args);
+  for (const auto& category_values : Kokkos::Impl::metadata_map) {
+    for (const auto& key_value : category_values.second) {
+      Kokkos::Tools::declareMetadata(key_value.first, key_value.second);
+    }
+  }
 }
-
+#define KOKKOS_METADATA_QUOTE_HELPER(x) #x
+#define KOKKOS_METADATA_QUOTE(x) KOKKOS_METADATA_QUOTE_HELPER(x)
 void pre_initialize_internal(const InitArguments& args) {
   if (args.disable_warnings) g_show_warnings = false;
   if (args.tune_internals) g_tune_internals = true;
+  std::stringstream str_builder;
+  str_builder << KOKKOS_VERSION / 10000 << "." << (KOKKOS_VERSION % 10000) / 100
+              << "." << KOKKOS_VERSION % 100;
```

Comment:
Maybe, refactor this into a separate function as well?

Reply:
I tried to make minimal changes, but sure

Reply:
577c2f0